src/React.js

```diff
@@ -65,5 +65,6 @@ assign(React, {
 });
 
 React.__SECRET_DOM_DO_NOT_USE_OR_YOU_WILL_BE_FIRED = ReactDOM;
+React.__SECRET_DOM_SERVER_DO_NOT_USE_OR_YOU_WILL_BE_FIRED = ReactDOMServer;
```

Comment:
`SECRET_DO_NOT_USE_OR_YOU_WILL_BE_FIRED` made sense when we were migrating and you really would break stuff.  But if we're adding this as an explicit hook to allow various environments, it shouldn't be a severe variable name (IMHO).  cc @zpao @spicyj for thoughts.